core/main.go

```diff
@@ -35,7 +36,7 @@ func initDatabases(cfg *config.Config) (*database.ServiceDB, *database.LevelDBIn
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	serviceDB := database.NewServiceDB(serviceStore)
+	serviceDB := database.NewServiceDB(serviceStore, json.Marshal, json.Unmarshal)
```

Comment:
we shouldn't pass the marshaler and unmarshaler here. It can be coded directly in the database. We will use only one encoding/decoding so it shouldn't be in the public api.

Reply:
it's because currently there is 2 encoding function used: one for the goleveldb, one with cosmos..
I cannot find of a nicer way..
Maybe two `NewServiceDB` functions? one for deprecated and one for cosmos?

Reply:
otherwise, what about simply removing the database package? the sdk could simply write directly into the underlying store without duplication of code.